avm/res/recovery-services/vault/backup-fabric/protection-container/main.bicep

```
@@ -0,0 +1,86 @@
+metadata name = 'Recovery Services Vault Protection Container'
+metadata description = 'This module deploys a Recovery Services Vault Protection Container.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Conditional. The name of the parent Azure Recovery Service Vault. Required if the template is used in a standalone deployment.')
+param recoveryVaultName string
+
+@description('Required. Name of the Azure Recovery Service Vault Protection Container.')
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Backup management type to execute the current Protection Container job.')
+@allowed([
+  'AzureBackupServer'
+  'AzureIaasVM'
+  'AzureSql'
+  'AzureStorage'
+  'AzureWorkload'
+  'DPM'
+  'DefaultBackup'
+  'Invalid'
+  'MAB'
+  ''
+])
+param backupManagementType string = ''
+
+@description('Optional. Resource ID of the target resource for the Protection Container.')
+param sourceResourceId string = ''
+
+@description('Optional. Friendly name of the Protection Container.')
+param friendlyName string = ''
+
+@description('Optional. Protected items to register in the container.')
+param protectedItems array = []
+
+@description('Optional. Type of the container.')
+@allowed([
+  'AzureBackupServerContainer'
+  'AzureSqlContainer'
+  'GenericContainer'
+  'Microsoft.ClassicCompute/virtualMachines'
+  'Microsoft.Compute/virtualMachines'
+  'SQLAGWorkLoadContainer'
+  'StorageContainer'
+  'VMAppContainer'
+  'Windows'
+  ''
+])
+param containerType string = ''
+
+resource protectionContainer 'Microsoft.RecoveryServices/vaults/backupFabrics/protectionContainers@2023-01-01' = {
+  name: '${recoveryVaultName}/Azure/${name}'
+  properties: {
+    sourceResourceId: !empty(sourceResourceId) ? sourceResourceId : null
+    friendlyName: !empty(friendlyName) ? friendlyName : null
+    backupManagementType: !empty(backupManagementType) ? backupManagementType : null
+    containerType: !empty(containerType) ? any(containerType) : null
+  }
+}
+
+module protectionContainer_protectedItems 'protected-item/main.bicep' = [for (protectedItem, index) in protectedItems: {
+  name: '${uniqueString(deployment().name, location)}-ProtectedItem-${index}'
+  params: {
+    policyId: protectedItem.policyId
+    name: protectedItem.name
+    protectedItemType: protectedItem.protectedItemType
+    protectionContainerName: name
```

```suggestion
    protectionContainerName: protectionContainer.name
```
Creates an implicit dependency through which we can get rid of the explicit dependency below